csv/write.go

```diff
@@ -7,8 +7,17 @@ import (
 	"github.com/cloudquery/plugin-sdk/schema"
 )
 
-func WriteTableBatch(w io.Writer, _ *schema.Table, resources [][]any) error {
+func WriteTableBatch(w io.Writer, table *schema.Table, resources [][]any, headers bool) error {
```

Comment:
I can't come up with any other way of injecting this conditional `headers` argument... Suggestions would be appreciated, or I can just add a nolint to it...

Reply:
I think they basically want you to split it into different functions, e.g. `WriteTableBatch` and `WriteTableBatchNoHeaders`. It might be better to force users to think about whether they want headers or not though, so I'm fine with adding `nolint` for this case

Reply:
I tried that... It still gave me the error...

Reply:
So I will just add the `nolint`

Reply:
I don't see how that can be possible, unless you still had a private function with the header boolean? In that case you could rather have `WriteTableBatch` write the headers first, then call `WriteTableBatchNoHeaders` to do the rest

Reply:
(Or something to that effect - probably won't work just like that) In any case, nolint is fine.

Reply:
Thinking about this some more... the function `WriteTableBatch(w io.Writer, table *schema.Table, resources [][]any) ` is a standard interface used throughout many destinations, and the plugin sdk... Maybe we need a more generic way of passing in arguments...

Reply:
So maybe it is better to add an entirely new function than it is to break that function signature...

Reply:
Yeah, though I think the filetypes don't need to conform to that interface. (CC @yevgenypats because he'll probably know better)

Reply:
A generic way of passing arguments sounds good though, as we'll hopefully be adding support for more formats (like parquet) in the not-too-distant future as well.

Reply:
I feel the current approach in this PR is the right one as there is no generic interface needed here and the caller need to know wheater they want to use headers or not.

Un related to that if we already add options to that table I would make this just a bit nicer with ...Option so we can extend this later without breaking the interface (For example, if we would want an option of changing a delimiter ).